### ISSUE TITLE
Error when using attribute-selector in combination with @At-root directive

### ISSUE THREAD

**Issue by siiron** [sasstools/sass-lint#786]
I use [gulp-sass-lint](https://github.com/sasstools/gulp-sass-lint) in my project and originally filed an [issue in that repo](https://github.com/sasstools/gulp-sass-lint/issues/50#issuecomment-232305378). The author asked me to file the issue here instead since gulp-sass-lint just consumes sass-lint.

So in context of my gulp-sass-lint setup, I'm currently on version 1.8.2 of sass-lint.

It seems that sass-lint reports a problem when using an attribute selector in combination with the @at-root directive when using scss syntax.

``` sass
.foo {
  border: 1px solid red;

  @at-root {
    // Sass-lint reports an error regarding the validity of the block"
    input[type=email]#{&} {
      border: 1px solid blue;
    }
  }
}
```

This is valid scss when using Libsass v3.3.6 and results into the following compiled CSS:

``` css
.foo {
  border: 1px solid red;
}

input[type=email].foo {
  border: 1px solid blue;
}
```

Using just a tag without attribute-selector seems to work fine without sass-lint reporting any errors:

``` sass
.bar {
  border: 1px solid green;

  @at-root {
    // Works fine
    input#{&} {
      border: 1px solid orange;
    }
  }
}
```

I'd expect sass-lint to not report the first block as an error.

Here's a [gist on Sassmeister](http://www.sassmeister.com/gist/31fff192b3131bc63737323ccfa4a438).

**Comment by DanPurdy** [comment]
Thanks @siiron it's just easier for us to keep track of these issues in sass-lint. It's a problem with the AST we use, Gonzales-pe whenever you see that `check the validity` message. Will look into it. 👍

**Comment by bgriffith** [comment]
This is fixed in the newly released version v1.11.0. Please let me know if you still have issues. Thanks!
